File: Regular Expressions/Code/chapter6/6-21/recipe6-21.pl
#!/usr/bin/perl -w
use strict;

open ( FILE, $ARGV[0] ) || die "Cannot open file!";

while (<FILE>)
{ 
	my $line = $_;
	# Jul 25 10:30:09 rhes-pe2400-01 pppd[1396]: local  IP address 209.98.170.206
	if ( $line =~ s/^.+\s+pppd\[\d+\]:\s+local\s+IP\s+address\s+([\d.]+)$/$1/ )
	{
		print $line;
	}
}

close( FILE );

File: Regular Expressions/Code/chapter6/6-21/recipe6-21.py
#!/usr/bin/python
import re
import sys

nargs = len(sys.argv)

if nargs > 1:

	myfile = sys.argv[1]
	output = open( myfile )
	regex = re.compile( r'^.+\s+pppd\[\d+\]:\s+local\s+IP\s+address\s+([\d.]+)$' )

	lines = output.readlines()
	for line in lines:
		if regex.match( line ):
			formatted = regex.sub( r'Local IP Address is:  \1', line )
			print formatted,

	output.close()

else:
	print 'Please supply a parameter!',
